src/Layout/Page/Page.component.tsx

```diff
@@ -2,50 +2,215 @@
-    background-color: ${colors.silver.base};
-    max-width: 100vw;
-    align-self: stretch;
-    padding: ${sizes.padding.lg} ${sizes.padding.xl};
+const StyledContent = styled.div<StyledContentProps>`
+    max-width: ${props => props.contentWidth};
+    background-color: ${props => props.contentBackgroundColor};
+    margin: 0 auto;
 `;
 
+// Interfaces
+// ------------------------------------------------------------------------------------------------------------------
 interface PageProps {
+    /** The header of the website. This can be a component. */
+    header?: any;
+    /** The footer of the website. This can be a component. */
+    footer?: any;
+    /** The height of the header. It will not hide on overflow. */
+    headerHeight?: string | number;
+    /** The height of the footer. It will not hide on overflow. */
+    footerHeight?: string | number;
+    /** Additional classname. */
+    className?: string;
+    children?: any;
+}
+
+interface DefaultLayoutProps {
+    /** The width of the layout, edge to edge. */
+    layoutWidth?: string | number;
+    /** The width of the content within the layout. */
+    contentWidth?: string | number;
+    /** The background color of the layout. Default is transparent. */
+    layoutBackgroundColor?: string;
+    /** The background color of the content within the layout area. Default is transparent. */
+    contentBackgroundColor?: string;
+    /** Additional classname. */
+    className?: string;
+    children?: any;
+}
+
+interface SidebarLayoutProps {
+    /** The width of the layout, edge to edge. */
+    layoutWidth?: string | number;
+    /** The width of the content within the layout. */
+    contentWidth?: string | number;
+    /** The width of the area for the sidebar. */
+    sidebarWidth?: string | number;
+    /** If the sidebar is on the left or the right of the layout */
+    sidebarAlign?: SidebarAlignment;
+    /** The actual sidebar. This can be a component. */
+    sidebar?: any;
+    /** The background color of the layout. Default is transparent. */
+    layoutBackgroundColor?: string;
+    /** The background color of the content within the layout area. Default is transparent. */
+    contentBackgroundColor?: string;
+    /** Additional classname. */
     className?: string;
     children?: any;
 }
 
-export const Page = ({ children }: PageProps): React.ReactElement<any> => (
-    <StyledPageElement className="anchor-page">
+interface StyledLayoutProps {
+    layoutWidth?: string | number;
+    layoutBackgroundColor?: string;
+}
+
+interface StyledContentProps {
+    contentWidth?: string | number;
+    contentBackgroundColor?: string;
+}
+
+// Components
+// ------------------------------------------------------------------------------------------------------------------
+export const Page = (props: PageProps): React.ReactElement<any> => (
+    <StyledPage className={classnames('anchor-page', props.className)}>
         <NormalizeCSS />
         <GlobalCSS />
```

Comment:
I'm debating on having `GlobalCSS`. What are your thoughts on this?

Reply:
I did most of the work on this w/o it, so I'm fine with ditching it. One possibility, a prop to enable it?

Reply:
I don't think it'll define anything anymore TBH. I created a Typography component that renders all permutations of our type, so we no longer need to have typography in there and that's all that was there. The Normalize is probably the only thing we need and IDK if that will be essential as we build out components with known un-normalized CSS we can just build that into the component.